src/binding/class.rs

```diff
@@ -127,7 +138,14 @@ pub fn freeze(object: Value) -> Value {
 }
 
 pub fn is_eql(object1: Value, object2: Value) -> Value {
-    unsafe { class::rb_eql(object1, object2) }
+    let result = unsafe { class::rb_eql(object1, object2) };
+    // In 3.1 and earlier we get Qtrue/ Qfalse
+    if cfg!(ruby_lte_3_1) {
+        result.into()
+    } else {
+        // After 3.1 we get TRUE/true
+        bool_to_value(c_int_to_bool(result))
+    }
```

Comment:
Can we have a test added for this as ruby version specifics have been added to it?  If it changed now then perhaps it changes in the future.

Reply:
The comment `// After 3.1 we get TRUE/true` says nothing of false.

Reply:
No sure if this is what you meant @danielpclark but I added a small test in here to detect if because of changes in the implementation of `rb_eql` things break at this level. The CI runs for all these ruby versions so we should be able to pin point which version is the offender.